src/custom/hooks/useAllTransactionsAndOrders.ts

```diff
@@ -0,0 +1,59 @@
+import { useMemo } from 'react'
+import { useAllTransactions } from 'state/transactions/hooks'
+import { useOrders } from 'state/orders/hooks'
+import { useActiveWeb3React } from 'hooks'
+import { Order, OrderStatus } from 'state/orders/actions'
+import { TransactionDetails } from '@src/state/transactions/reducer'
+
+type TransactionAndOrder =
+  | (Order & { addedTime: number })
+  | (TransactionDetails & {
+      id: string
+      status: 'pending' | 'fulfilled'
+    })
+
+/**
+ * Returns whether a transaction/order happened in the last day (86400 seconds * 1000 milliseconds / second)
+ * @param time to check for recency
+ */
+export function isRecent(time: number): boolean {
+  return new Date().getTime() - time < 86_400_000
+}
+
+function isPending(data: TransactionAndOrder) {
+  return data.status === OrderStatus.PENDING
+}
+
+function isFulfilled(data: TransactionAndOrder) {
+  return data.status === OrderStatus.FULFILLED
+}
+
+export default function useAllTransactionsAndOrders() {
+  const { chainId } = useActiveWeb3React()
+  const allTransactions = useAllTransactions()
+  const allNonEmptyOrders = useOrders({ chainId })
+
+  return useMemo(() => {
+    const adjustedTransactions = Object.values(allTransactions)
+      .map(tx => ({ ...tx, id: tx.hash, status: tx.receipt ? 'fulfilled' : 'pending' }))
+      .filter(tx => isRecent(tx.addedTime))
```

Comment:
Is this part of the old codes logic? How did you come up with the constant and all?
Shouldn't this be mentioned in the PR description?

Reply:
what constant? this logic is to have one array of pending "activities" as i call them

Reply:
I'm referring to the logic of `isRecent`. There's a constant involved in that, and a logic you are doing here of filtering out old transactions

My question is, do they do this to? what is the criteria? why is this not commented in the description, I would consider it important for the review (it looks like a functional decision)

Reply:
Also, and independently of this, shouldn't this filtering be handled with an action. It feels overcomplicated this parts